### PR TITLE
Fix: Smart shipping defaults bug

### LINKED ISSUE #34319: Shipping Methods > Prevent already set shipping methods from being overriden by Onboarding screens (can lead to margin lost)
### Prerequisites

- [x] I have carried out troubleshooting steps and I believe I have found a bug.
- [x] I have searched for similar bugs in both open and closed issues and cannot find a duplicate.

### Describe the bug

It is hard to get rid of Onboarding process "panel" in WP Dashboard. It proposed too many things (extra plugins)
But since the onboarding process is not closed, a user can still play with it but it can override / change your settings.
Adding automatically free shipping methods can lead to extra owner costs ! (in each case, all order are free of delivery costs !)
I've finally found a way with "hide" panel with the 3 dots, and i did not see again the "6/6 Start to sell".

### Expected behavior

Secure the onboarding by preventing it to override already set settings.
Check if personal methods have been already set before adding default free shipping method
Ease onboarding by proposing later on the none critical plugins
All marketing stuffs are not needed to start a store

### Actual behavior

see https://github.com/woocommerce/woocommerce/blob/e05f52b18a2054488238524afb1a7566cdfe7b9a/plugins/woocommerce/src/Internal/Admin/Homescreen.php#L128-L131

### Steps to reproduce

1 Set your own shipping methods
2. Each time you "play" with a not closed onboarding process (smart screens), a default free shipping zone is added (set on the shop country area), even if you already have set your own methods.

### WordPress Environment

Woo 6.7/6.8

### Isolating the problem

- [X] I have deactivated other plugins and confirmed this bug occurs when only WooCommerce plugin is active.
- [X] This bug happens with a default WordPress theme active, or [Storefront](https://woocommerce.com/storefront/).
- [X] I can reproduce this bug consistently using the steps above.

### PR DESCRIPTION
**Pull request by moon0326**
### All Submissions:

-   [X] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [X] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [X] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Closes #34319 

This PR fixes smart shipping defaults bug.

As noted in the issue, we should not create a new shipping zone when there is already one.

This PR counts # of shipping zones to make sure we don't create/override a shipping zone.

### How to test the changes in this Pull Request:

1. Start with a fresh install.
2. Do not initiate OBW yet.
3. Go to `WooCommerce -> Settings -> Shiping` and create a shipping zone.
4. Complete OBW and choose a country not that is not one of the US, CA, AU, or GB.
5. Navigate to `WooCommerce -> Home`
6. Navigate to `WooCommerce -> Settings -> Shiping` and confirm no extra shipping zone exist. 
7. Confirm `woocommerce_admin_created_default_shipping_zones` option has been created.

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you successfully run tests with your changes locally?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm changelog add --filter=<project>`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

File: plugins/woocommerce/src/Internal/Admin/Homescreen.php [hunk 1]
```diff
@@ -90,6 +90,12 @@ public function maybe_set_default_shipping_options_on_home( $settings ) {
 			return $settings;
 		}
 
+		$zone_count = count( \WC_Data_Store::load( 'shipping-zone' )->get_zones() );
+		if ( $zone_count ) {
+			update_option( 'woocommerce_admin_created_default_shipping_zones', 'yes' );
```
Comment (ilyasfoo) [inline plugins/woocommerce/src/Internal/Admin/Homescreen.php:95]: Good call setting the option here!

**Review by ilyasfoo — DISMISSED** [review]
Tested well, thanks for the fix, @moon0326! LGTM 🚢

**Review by ilyasfoo — APPROVED** [review]

**Comment by rohderr** [comment]
Due to this mistake on Woo's part, we lost $6,000 as a "Free Shipping" zone was added silently by Woo, and customers did not hesitate to take advantage of that.  

https://wordpress.org/support/topic/woo-update-added-free-shipping-zone-without-consent/#post-15925644

**Comment by DanHoelck** [comment]
> Due to this mistake on Woo's part, we lost $6,000 as a "Free Shipping" zone was added silently by Woo, and customers did not hesitate to take advantage of that.

Had the same problem on our site. Not great.

**Comment by davideprevosto** [comment]
> Due to this mistake on Woo's part, we lost $6,000 as a "Free Shipping" zone was added silently by Woo, and customers did not hesitate to take advantage of that.
> 
> https://wordpress.org/support/topic/woo-update-added-free-shipping-zone-without-consent/#post-15925644

It's sounds like a complain: understandable. However we are not obliged to use a free, open source product for our e-commerce projects. #2cents

**Pull request merged**